lib/Onyx.js

```diff
@@ -35,6 +36,39 @@ let defaultKeyStates = {};
 // Connections can be made before `Onyx.init`. They would wait for this task before resolving
 const deferredInitTask = createDeferredTask();
 
+/**
+ * Uses a selector string or function to return a simplified version of sourceData
+ * @param {Mixed} sourceData
+ * @param {String|Function} selector
+ *      If it's a string, the selector is passed to lodashGet on the sourceData
+ *      If it's a function, it is passed the sourceData and it should return the simplified data
+ * @returns {Mixed}
+ */
+const getSubsetOfData = (sourceData, selector) => (_.isFunction(selector)
+    ? selector(sourceData)
+    : lodashGet(sourceData, selector));
+
+/**
+ * Takes a collection of items (eg. {testKey_1:{a:'a'}, testKey_2:{b:'b'}})
+ * and runs it through a reducer function to return a subset of the data according to a selector.
+ * The resulting collection will only contain items that are returned by the selector.
+ * @param {Object} collection
+ * @param {String|Function} selector (see method docs for getSubsetOfData() for full details)
+ * @returns {Object}
+ */
+const reduceCollectionWithSelector = (collection, selector) => _.reduce(collection, (finalCollection, item, key) => {
+    const selectedValue = getSubsetOfData(item, selector);
+
+    // A selector is used to return a subset of a larger dataset.
+    // If the selector didn't return any value, then nothing is added to the finalCollection.
+    if (!_.isUndefined(selectedValue)) {
+        // eslint-disable-next-line no-param-reassign
+        finalCollection[key] = selectedValue;
+    }
```

Comment:
I will go ahead and change this to allow returning undefined. It actually doesn't really change anything in the implementation that I can tell (it at least doesn't break any of the tests).

Reply:
I guess that wasn't totally true. I had to change the selector from `obj => obj.ja` to `obj => obj && obj.a` since it can now be passed an undefined object.